feat: add accessibility

jaaywags opened the pull request:
Added accessibility for screen readers.

Tested and verified it works with

- [x] The built in iOS screen reader
- [ ] The built in Android screen reader (still testing)

jaaywags commented:
This will resolve issue #71

jaaywags commented:
I am probably going to go back in and add a property, `accessibilityToggleSize` (or some other name). That way they can control how much the `stagedRating` changes on increment and decrement actions. That way it is not always 1. I had someone complain already that they wanted to do half stars. Though, it will default to 1.

I am also thinking about allow the user to pass in properties for the accessibility labels. That way they can modify the language if they want.

bviebahn commented:
Thanks, this looks great! I was just trying to test the behaviour of the rating component in the iOS app store to get some inspiration, but I'm struggling a bit to use the the actions... I'll try again a bit later when I have some more time. Have you tested it? I'm also wondering if incrementing/decrementing should change the value directly, instead of having to apply it using the activate action. This way you would still get the visual feedback of the stars changing. What do you think?

jaaywags commented:
> Thanks, this looks great! 
🎉

>  I was just trying to test the behaviour of the rating component in the iOS app store
Were you having trouble figuring out the screen reader actions? I can post a quick guide if that helps.

I was actually using the app store star rating as inspiration too. I had copied their accessibility label and actions. They use the stage method which is why I used it but I am totally okay not doing that and just applying the rating. I think that is how it was on Android. They use a slider method and as you slide it applies that new rating immediately.

> Have you tested it?
I tested this update on my app for both iOS and Android.

bviebahn commented:
Now I figured it out. Ok let's keep it like this then! I'm wondering if instead of adding the `accessibilityToggleSize ` prop we should just also increment/decrement half steps if `enableHalfStar` is true. Otherwise users who use accessibility features are a little more limited than users who don't, which is not very accessible, right? Otherwise we would also have to handle the case that the rating is already set at a half value. Currently if it starts at 2.5 for example you can only increment to 3.5.

jaaywags commented:
> I'm wondering if instead of adding the accessibilityToggleSize  prop we should just also increment/decrement half steps if enableHalfStar is true

I 100% agree! For some reason I thought you could customize the increment size to any decimal but I think that was another star library I was looking at.

I can make that update to check for that `enableHalfStar` flag instead. That would be much better.

I probably won't be able to make the change until Monday/Tuesday though. I am visiting family and trying to limit my computer time lol.

> Otherwise users who use accessibility features are a little more limited than users who don't,

You are correct! And I already got that feedback too on my app. It should be the same for everyone.

bviebahn commented:
Sure, take your time. Enjoy your weekend :)

jaaywags commented:
@bviebahn I think I got everything.

It now does half stars if the `enableHalfStar` flag is true.

I also added additional properties to allow the user to customize the accessibility labels. I was thinking this would help support other languages.

I also updated the readme.

Let me know if you want me to change anything :)

bviebahn commented:
Looks perfect! I just created a new release 1.9.0. Thanks for your help! 🚀
